plugin/src/main/java/me/block2block/hubparkour/listeners/PressurePlateListener.java

```diff
@@ -94,28 +95,42 @@ public void run() {
         }
         if (CacheManager.isParkour(e.getPlayer())) {
             if (e.getTo().getBlock().isLiquid()) {
+                boolean tpwater = ConfigUtil.getBoolean("Settings.Teleport.On-Water", true);
+                boolean tplava = ConfigUtil.getBoolean("Settings.Teleport.On-Lava", true);
+                ParkourPlayerTeleportEvent.TeleportReason reason =
+                        ParkourPlayerTeleportEvent.TeleportReason.Unknow;
                 if (e.getTo().getBlock().getType() == Material.WATER) {
-                    if (!ConfigUtil.getBoolean("Settings.Teleport.On-Water", true)) {
+                    reason = ParkourPlayerTeleportEvent.TeleportReason.Water;
+                    if (!tpwater) {
                         return;
                     }
                 } else if (e.getTo().getBlock().getType() == Material.LAVA) {
-                    if (!ConfigUtil.getBoolean("Settings.Teleport.On-Lava", true)) {
+                    reason = ParkourPlayerTeleportEvent.TeleportReason.Lava;
+                    if (!tplava) {
                         return;
                     }
                 } else if (HubParkour.isPre1_13()) {
                     if (e.getTo().getBlock().getType() == Material.getMaterial("STATIONARY_WATER")) {
-                        if (!ConfigUtil.getBoolean("Settings.Teleport.On-Water", true)) {
+                        reason = ParkourPlayerTeleportEvent.TeleportReason.Water;
+                        if (!tpwater) {
                             return;
                         }
                     } else {
-                        if (!ConfigUtil.getBoolean("Settings.Teleport.On-Lava", true)) {
+                        reason = ParkourPlayerTeleportEvent.TeleportReason.Lava;
+                        if (!tplava) {
                             return;
                         }
                     }
                 }
-                if (ConfigUtil.getBoolean("Settings.Teleport.On-Water", true)) {
-                    HubParkourPlayer player = CacheManager.getPlayer(e.getPlayer());
```

Comment:
Removing the already existing player object is pointless, as it means you are making 2 calls to CacheManager#getPlayer.